.github/workflows/pull-request-checks.yaml

```diff
@@ -11,14 +11,84 @@ on:
       - master
 
 jobs:
-  build:
+  check_version:
+    name: Ensure component commits modify component versions
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2.4.0
+      name: Checkout repo
+      with:
+        # See https://github.com/actions/checkout#checkout-v2
+        # This will be slow. The intent is to fetch all commits
+        # since the merge-base (the commit where we branched off)
+        # so we can check the git diff against all changed files.
+        # By default, the checkout action only returns the last commit,
+        # There's no native way to do this in the checkout action, so 
+        # we have to fetch the entire history. See
+        # https://github.com/actions/checkout/issues/266#issuecomment-638346893
+        fetch-depth: 0
+    - uses: jitterbit/get-changed-files@v1
+      id: changed_files
+      name: Get changed files
+    # We need to get a diff of the base and HEAD commit below
+    - name: Return base commit
+      id: base_commit
+      uses: actions/github-script@v6
+      with:
+        script: return context.payload.pull_request?.base?.sha
+        result-encoding: string
+    - name: Return HEAD commit
+      id: head_commit
+      uses: actions/github-script@v6
+      with:
+        script: return context.payload.pull_request?.head?.sha
+        result-encoding: string
+    # Diff the changes in the HEAD ref (new branch) against the BASE ref (target branch, e.g. master)
+    # If the changes don't contain modifications to the version, we want to log that fact but not exit early
+    # Once we've checked all files, exit the script if any files failed to modify version
+    - id: git_diff_on_components
+      name: Check git diff for version changes
+      run: |-
+        commit_contained_components_that_did_not_modify_version=0
+        for f in ${{ steps.changed_files.outputs.all }}
+        do
+          ext="${f##*.}"
+          # Only run this check on modified sources or actions, excluding common files
+          if ([[ "$ext" == "js" ]] || [[ "$ext" == "mjs" ]]) && \
+             ([[ "${f}" == *components/**/sources/* ]] || [[ "${f}" == *components/**/actions/* ]]) && \
+             [[ "${f}" != *common*.*js ]] && [[ "${f}" != **/actions/common/* ]] && [[ "${f}" != **/sources/common/* ]]
```

Comment:
I think this condition to include only component files should work for almost every file.
**Minor issue**:
There are a small number of **non-component** files in the repo that pass this condition, which would cause the check to fail. For example, these files pass the `if` condition, but don't contain `version:` (in the diff):
- [components/datadog/sources/payload-format.js](https://github.com/PipedreamHQ/pipedream/blob/f38ef1994a174fe51c3df4c6c3971a4357de9395/components/sendgrid/sources/events/engagement-event-types.js)
- [components/sendgrid/sources/events/delivery-event-types.js](https://github.com/PipedreamHQ/pipedream/blob/f38ef1994a174fe51c3df4c6c3971a4357de9395/components/sendgrid/sources/events/delivery-event-types.js)

**Potential fix**:
Add a condition to the `if` statement which checks if the file contains "version:", so that files without "version:" are excluded from the check. E.g.,
```sh
             && [[ "$(<${f})" == *"version:"* ]]
```
Drawback: Component files missing a "version:" will be skipped. (I think this should rarely be an issue because components require a `version` property to be published via the CLI.)

**Other potential fixes**:
- Require, as a guideline, that new/updated non-component JS files, in the `source` or `action` folders, be nested in the `common` folder
- Improve component detection (the `if` condition) by checking features/properties of the file (e.g., an object as `default export`, inclusion of `key`, `type`, `version` properties, etc.)
- For now, ignore failed checks for non-component files if they occur, since they should be uncommon